src/components/PostPageDetail/index.js

```diff
@@ -112,12 +112,11 @@ const PostPageDetailIdComponent = (props) => {
   };
   const initial = () => {
     const reactionCount = item?.reaction_counts;
-    if (JSON.stringify(reactionCount) !== '{}') {
+    if (JSON.stringify(reactionCount) !== '{}' && reactionCount) {
       let count = 0;
-      const {comment} = reactionCount;
       handleVote(reactionCount);
-      if (comment !== undefined) {
-        if (comment > 0) {
+      if (reactionCount?.comment !== undefined) {
+        if (reactionCount?.comment > 0) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of optional chaining (`?.`) is a good practice to avoid potential `undefined` errors. However, you are checking twice if `reactionCount?.comment` is not `undefined`. You can simplify this by merging the two conditions into one.
```diff
-      if (reactionCount?.comment !== undefined) {
-        if (reactionCount?.comment > 0) {
+      if (reactionCount?.comment > 0) {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->